DEVELOPING.md

```diff
@@ -0,0 +1,51 @@
+# Development guides and tips  
+
+## Before Posting PR
+
+Developers are expected to run the following prior to posting PRs.  
+
+```shell
+make ci-job
+```
+
+This will run the various linters, as well as required unit tests.  
+
+## Linting
+
+### Run `golangci-lint`
+
+Full list linters and config can be found in `.golangci.yml`
+
+- The golangci-lint test can be run individually at any time during development (Note: `make ci-job` also runs this)
+
+  ```shell
+  make golangci-lint
+  ```
+
+- Or run the following to allow `golangci-lint` for any auto fixes (if available by linter)
+  
+  ```shell
+  golangci-lint run --fix --config .golangci.yml
+  ```
+
+### Suppress `golangci-lint` errors
+
+If you would like to suppress a false positive, consider in-line syntax. Please note that such suppressions should only be used in exceptional cases, however, rather than common practice.
+
+An example error for gosec
+
+```shell
```

Comment:
very minor nit: Should be console, rather than shell

Reply:
updated